### DISCUSSION TITLE
c++ template

### DISCUSSION THREAD

**Discussion by softopti** [alefragnani/vscode-separators#102] [Q&A] [answered]
Is it possible to have the separator before the keyword template in c++ ?

**Comment by alefragnani** [comment, marked as answer]
Hi @softopti ,

It’s been a looong time I don’t play with C++, and I’m not aware of the `template` keyword, so I don’t even have any C++ extension installed to try out. But, either way, I would say, it depends.

The extension depends entirely on the language extension in order to identify the _symbols_, and to know if some _symbol_ could receive separators, you could use the `Outline` view as a reference. If you see the `templates` like classes, functions and so on, `templates` could certainly receive separators above. On the other hand, if `templates` are not in the `Outline` view, so the extension won’t be able to draw separators.

If you see them in the `Outline` view, please post a picture of it, and if possible, a sample source that I could use play with.

Hope this helps
